### ISSUE TITLE
Stop using internal require('module')._preloadModules

### ISSUE THREAD

**Issue by BridgeAR** [electron/electron#17780]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description

Electron currently uses the "officially" Node.js "internal" `_preloadModules` function. https://github.com/electron/electron/blob/9f30e5f52635c797f633fbbc1c4417f1fad5664b/atom/browser/default_app/main.js#L219

It is not desirable to use internal functions and there is no other frequently used module that uses this function. Therefore I would like to make change it to be completely private and not expose this to userland anymore. Is there any possibility to change the logic here to just pass through the options so that the preloading would still work? 

### Proposed Solution

Change the the implementation to passing through `-r foobar` while starting Node.js.

**Comment by MarshallOfSound** [comment]
I seem to recall with the way we start node internally it doesn't get any of the cli args or something like that.  Any thoughts on the above cc @codebytere

**Comment by codebytere** [comment]
🤔 this is a good question & point @BridgeAR, i'll dig into this and see what we can do to enable you making those changes in node core!

**Comment by codebytere** [comment]
Yeah so we load the node env in `PostEarlyInitialization`, way before any code is executed so we wouldn't be able to pass these args that way at first glance. Would Node consider exposing a public API for this @BridgeAR ?

**Comment by BridgeAR** [comment]
@codebytere do you actually load any javascript on your side before starting the user application? If so, the preload could also just be done in there instead by calling require on the modules that should be preloaded.

Or can you outline how the bootstrapping is done in electron currently?

I guess we could make it public but it should be super rare that anyone wants to hook into that (if the Node.js bootstrapping is already done and JS is executed, then there's no point in using the preload anymore).

**Comment by nornagon** [comment]
Just checking—is this still something we want to address or can we close this out?

**Comment by felixrieseberg** [comment]
I'm going to close this issue for now, it's been a bunch of years - and we can always come back to this once someone feels like it should be implemented.
